### PR TITLE
Coral-Trino: Add type casting for concat function

### PR DESCRIPTION
**Pull request by ljfgem**
This PR is to solve #247

### CODE REVIEW

File: coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/Calcite2TrinoUDFConverter.java [hunk 1]
```diff
@@ -255,6 +262,24 @@ public RexNode visitCall(RexCall call) {
...
 
+    private Optional<RexNode> visitConcat(RexCall call) {
+      // Hive supports operations like CONCAT(date, varchar) while Trino only supports CONCAT(varchar, varchar)
+      // So we need to cast the unsupported types to varchar
+      final SqlOperator op = call.getOperator();
+      List<RexNode> convertedOperands = visitList(call.getOperands(), (boolean[]) null);
+      List<RexNode> castOperands = new ArrayList<>();
+
+      for (RexNode inputOperand : convertedOperands) {
+        if (inputOperand.getType().getSqlTypeName() != VARCHAR && inputOperand.getType().getSqlTypeName() != CHAR) {
```
Comment (aastha25) [inline coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/Calcite2TrinoUDFConverter.java:273]: will this condition always be false? can getType().getSqlTypeName() be both VARCHAR and CHAR?

can this be changed to - 
!Sets.newHashSet(SqlTypeName.VARCHAR, SqlTypeName.CHAR).contains(inputOperand.getType().getSqlTypeName())

**Review by aastha25 — COMMENTED** [review]
Thanks for the PR!

File: coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/Calcite2TrinoUDFConverter.java (hunk 1 above)
Comment (aastha25) [reply coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/Calcite2TrinoUDFConverter.java:273]: okay, yeah sorry my bad.
technically this implementation is also correct.

File: coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/Calcite2TrinoUDFConverter.java [hunk 2]
```diff
@@ -255,6 +262,24 @@ public RexNode visitCall(RexCall call) {
       return adjustReturnTypeWithCast(rexBuilder, super.visitCall(modifiedCall));
     }
 
+    private Optional<RexNode> visitConcat(RexCall call) {
+      // Hive supports operations like CONCAT(date, varchar) while Trino only supports CONCAT(varchar, varchar)
+      // So we need to cast the unsupported types to varchar
```
Comment (aastha25) [inline coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/Calcite2TrinoUDFConverter.java:267]: should we also validate that the "unsupported type" is a date? @wmoustafa

Comment (ljfgem) [reply coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/Calcite2TrinoUDFConverter.java:267]: I don't think that is necessary, the input type has been validated in `StaticHiveFunctionRegistry`, we just need to cast the type to be aligned with Trino's expectation.

Comment (aastha25) [reply coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/Calcite2TrinoUDFConverter.java:267]: Thanks for the clarification.

**Comment by aastha25** [comment]
Thanks for the PR @ljfgem  LGTM

**Pull request merged**